Tests/BotTests/GitHub/GitHubEventsTests.swift

```diff
@@ -2,83 +2,164 @@ import XCTest
 import Nimble
 import Result
 import ReactiveSwift
+import CryptoSwift
 @testable import Bot
 
 class GitHubEventsTests: XCTestCase {
 
     func test_handling_pull_request_event() {
-        let service = GitHubService()
-        let scheduler = TestScheduler()
+        perform(
+            when: { service, scheduler in
 
-        let pullRequestEvent = PullRequestEvent(
-            action: .opened,
-            pullRequestMetadata: PullRequestMetadata(
-                reference: PullRequest(
-                    number: 1,
-                    title: "Hello World",
-                    author: PullRequest.Author(login: "Wall-E Bot"),
-                    source: PullRequest.Branch(ref: "some-feature", sha: "123"),
-                    target: PullRequest.Branch(ref: "master", sha: "123"),
-                    labels: []
-                ),
-                isMerged: false,
-                mergeState: .clean
-            )
+                let req = request(
+                    with: .pullRequest,
+                    signature: "00faadd5dc1395b616a546ac6f67796409b4f839",
+                    payload: GitHubPullRequestEvent.data(using: .utf8)
+                )
+
+                let result = service.handleEvent(from: req).first()
+
+                scheduler.advance()
+
+                expect(result?.error).to(beNil())
+            },
+            assert: { events in
+                expect(events) == [Event.pullRequest(stubbedPullRequestEvent)]
+            }
         )
+    }
+
+    func test_handling_ping_event() {
+        perform(
+            when: { service, scheduler in
 
-        let request = StubbedRequest(
-            headers: [GitHubService.HTTPHeader.event.rawValue: GitHubService.APIEvent.pullRequest.rawValue],
-            body: pullRequestEvent
+                // NOTE: I'm unsure about the payload of this specific event
+                let req = request(
+                    with: .ping,
+                    signature: "9f64868b5cd91faa4c63589acff7286b16d289ae",
+                    payload: "{}"
+                )
+
+                let result = service.handleEvent(from: req).first()
+
+                scheduler.advance()
+
+                expect(result?.error).to(beNil())
```

Comment:
Shouldn't all those `expect` be grouped in the assert closure (which you'd tweak to also pass the result as parameter)?

Reply:
Probably makes sense, let's do it.